rotkehlchen/chain/ethereum/names.py

```diff
@@ -1,13 +1,63 @@
+import time
 from typing import Dict, List
 
 from eth_utils import to_checksum_address
 
 from rotkehlchen.chain.ethereum.decoding.constants import ETHADDRESS_TO_KNOWN_NAME
+from rotkehlchen.chain.ethereum.manager import EthereumManager
+from rotkehlchen.constants import ENS_UPDATE_INTERVAL
 from rotkehlchen.db.addressbook import DBAddressbook
 from rotkehlchen.db.dbhandler import DBHandler
 from rotkehlchen.db.ens import DBEns
+from rotkehlchen.errors.misc import BlockchainQueryError, RemoteError
 from rotkehlchen.globaldb import GlobalDBHandler
-from rotkehlchen.types import AddressbookType, ChecksumEvmAddress, EnsMapping, SupportedBlockchain
+from rotkehlchen.types import (
+    AddressbookType,
+    ChecksumEvmAddress,
+    EnsMapping,
+    SupportedBlockchain,
+    Timestamp,
+)
+
+
+def find_ens_mappings(
+        ethereum_manager: EthereumManager,
+        addresses: List[ChecksumEvmAddress],
+        ignore_cache: bool,
+) -> Dict[ChecksumEvmAddress, str]:
+    """
+    Find and return ens names for the given addresses.
+    First check the db, and if can't find, call the blockchain.
+    """
+    dbens = DBEns(ethereum_manager.database)
+    ens_mappings: Dict[ChecksumEvmAddress, str] = {}
+    with dbens.db.user_write() as cursor:
+        if ignore_cache:
+            addresses_to_query = addresses
+        else:
+            addresses_to_query = []
+            cached_data = dbens.get_reverse_ens(cursor, addresses)
+            cur_time = Timestamp(int(time.time()))
+            for address, cached_value in cached_data.items():
+                has_name = isinstance(cached_value, EnsMapping)
+                last_update: Timestamp = cached_value.last_update if has_name else cached_value  # type: ignore  # noqa: E501
```

Comment:
why the type ignore? When you add one always add a comment for why it is needed

Reply:
Honestly, no idea since I just moved this function from one place (`rest.py`) to another (separate file). But I will check